modules/dhcp_common/subnet.rb

```diff
@@ -62,7 +62,7 @@ def netmask_to_i
     end
 
     def validate_subnet_range!(from_address, to_address)
-      if (from = validate_ip(from_address)) && (to = validate_ip(to_address))
+      if (from = validate_ip(from_address, 4)) && (to = validate_ip(to_address, 4))
         raise Proxy::DHCP::Error, "Range does not belong to provided subnet" unless include?(from) && include?(to)
         from = IPAddr.new(from)
         to   = IPAddr.new(to)
```

Comment:
Please remove the changes to validate only for IPv4 from `modules/dhcp_common` files. See, these classes are shared by other plugins and may or may not support IPv6, we don't know (e..g. Infoblox does support it I think).

Where it is appropriate to do these checks are SP modules we know do not support IPv6, these are: MS native, libvirt and isc.

So please find `add_record` method and perform IPv4 only validation there.